src/Grpc.Net.Client/Internal/Retry/RetryCallBase.cs

```diff
@@ -542,7 +542,20 @@ protected void HandleUnexpectedError(Exception ex)
             else
             {
                 commitReason = CommitReason.Canceled;
-                resolvedCall = CreateStatusCall(Disposed ? GrpcProtocolConstants.CreateDisposeCanceledStatus(ex) : GrpcProtocolConstants.CreateClientCanceledStatus(ex));
+                Status status;
+                if (Disposed)
+                {
+                    status = GrpcProtocolConstants.CreateDisposeCanceledStatus(exception: null);
+                }
+                else
+                {
+                    if (Options.CancellationToken.IsCancellationRequested && Options.CancellationToken != operationCanceledException.CancellationToken)
```

Comment:
I just want to check the intent here... "if something cancelled (`ex`), and `Options` is *also* cancelled, but isn't the thing being reported by `ex`: replace `ex` with the cancellation of `Options`" - is that right? are we happy to simply lose the original `ex` (which is a *different* calcellation)? should it be an inner-exception?

Reply:
This is replacing the OCE from CancellationTokenSource with an OCE using Options.CancellationToken if it is canceled. I added a comment.

The original exception is from an internal CT so it isn't valuable to keep. An OCE nested inside an OCE would be confusing.